eclipse-collections/src/main/java/org/eclipse/collections/impl/set/mutable/UnifiedSet.java

```diff
@@ -773,6 +774,32 @@ else if (cur != null)
         return null;
     }
 
+    @Override
+    protected Optional<T> detectOptional(Predicate<? super T> predicate, int start, int end)
+    {
+        for (int i = start; i < end; i++)
+        {
+            Object cur = this.table[i];
+            if (cur instanceof ChainedBucket)
+            {
+                Object chainedDetect = this.chainedDetect((ChainedBucket) cur, predicate);
+                if (chainedDetect != null)
+                {
+                    return Optional.of(this.nonSentinel(chainedDetect));
+                }
+            }
+            else if (cur != null)
+            {
+                T each = this.nonSentinel(cur);
+                if (predicate.accept(each))
+                {
+                    return Optional.of(each);
+                }
+            }
+        }
+        return Optional.empty();
+    }
+
     private Object chainedDetect(ChainedBucket bucket, Predicate<? super T> predicate)
```

Comment:
Very nice, the return value of chainedDetect() is already unambiguous and only returns null to mean not found.